cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/template/etl/batch/sink/ElasticsearchSink.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.template.etl.batch.sink;
+
+import co.cask.cdap.api.annotation.Description;
+import co.cask.cdap.api.annotation.Name;
+import co.cask.cdap.api.annotation.Plugin;
+import co.cask.cdap.api.data.format.StructuredRecord;
+import co.cask.cdap.api.dataset.lib.KeyValue;
+import co.cask.cdap.api.templates.plugins.PluginConfig;
+import co.cask.cdap.template.etl.api.Emitter;
+import co.cask.cdap.template.etl.api.batch.BatchSink;
+import co.cask.cdap.template.etl.api.batch.BatchSinkContext;
+import co.cask.cdap.template.etl.common.Properties;
+import co.cask.cdap.template.etl.common.StructuredRecordStringConversion;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.io.Text;
+import org.apache.hadoop.io.Writable;
+import org.apache.hadoop.mapreduce.Job;
+import org.elasticsearch.hadoop.mr.EsOutputFormat;
+
+/**
+ * A {@link BatchSink} that writes data to a Elasticsearch.
+ * <p/>
+ * This {@link ElasticsearchSink} takes a {@link StructuredRecord} in,
+ * converts it to a json per {@link StructuredRecordStringConversion},
+ * and writes it to the Elasticserach server.
+ * <p/>
+ * If the Elasticserach index does not exist, it will be created using the default properties
+ * specified by Elasticsearch. See more information at
+ * https://www.elastic.co/guide/en/elasticsearch/guide/current/_index_settings.html.
+ * <p/>
+ */
+@Plugin(type = "sink")
+@Name("Elasticsearch")
+@Description("CDAP Elasticsearch Batch Sink")
```

Comment:
Should probably add a lot more description. Maybe some stuff from this class' javadoc.